cdflow_commands/plugins/aws_lambda.py

```diff
@@ -0,0 +1,228 @@
+        version
+    )
+    deploy_factory = build_deploy_factory(
+        environment_name,
+        component_name,
+        version,
+        metadata,
+        global_config,
+        root_session
+    )
+    return LambdaPlugin(
+        release_factory,
+        deploy_factory,
+        None
+    )
+
+
+def build_release_factory(
+    global_config, root_session, component_name, metadata, version
+):
+    def _release_factory():
+        boto_s3_client = assume_role(
+            root_session,
+            global_config.dev_account_id,
+            get_role_session_name(os.environ)
+        ).client('s3')
+        return Release(
+            global_config,
+            boto_s3_client,
+            component_name,
+            metadata,
+            version
+        )
+    return _release_factory
+
+
+def build_deploy_factory(
+    environment_name, component_name, version,
+    metadata, global_config, root_session
+):
+    def _deploy_factory():
+        is_prod = environment_name == 'live'
+        if is_prod:
+            account_id = global_config.prod_account_id
+        else:
+            account_id = global_config.dev_account_id
+
+        boto_session = assume_role(
+            root_session,
+            account_id,
+            get_role_session_name(os.environ)
+        )
+        deploy_config = DeployConfig(
+            team=metadata.team,
+            dev_account_id=global_config.dev_account_id,
+            platform_config_file=get_platform_config_path(
+                metadata.account_prefix,
+                metadata.aws_region,
+                is_prod
+            )
+        )
+        return Deploy(
+            boto_session,
+            component_name,
+            environment_name,
+            version,
+            deploy_config
+        )
+    return _deploy_factory
+
+
+class LambdaPlugin(Plugin):
+    def __init__(
+        self,
+        release_factory,
+        deploy_factory,
+        destroy_factory
+    ):
+        self.release_factory = release_factory
+        self.deploy_factory = deploy_factory
+
+    def release(self):
+        release = self.release_factory()
+        release.create()
+
+    def deploy(self):
+        deploy = self.deploy_factory()
+        deploy.run()
+
+    def destroy(self):
+        pass
+
+
+class Release():
+    def __init__(
+        self,
+        global_config,
+        boto_s3_client,
+        component_name,
+        metadata,
+        version
+    ):
+        self._global_config = global_config
+        self._boto_s3_client = boto_s3_client
+        self._component_name = component_name
+        self._metadata = metadata
+        self._version = version
+
+    def create(self):
+        zip = ZipFile(self._component_name + '.zip', 'x')
+
+        for dirname, subdirs, files in os.walk(self._component_name):
+            zip.write(dirname)
+            for filename in files:
+                zip.write(os.path.join(dirname, filename))
+        zip.close()
```

Comment:
It might be nice to use a context manager here so we don't have to worry about explicitly calling `close()` on the resource: https://docs.python.org/3/library/zipfile.html#zipfile.ZipFile.open